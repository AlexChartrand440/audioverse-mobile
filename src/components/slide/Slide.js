import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, Image, Text, StyleSheet } from 'react-native'

const Slide = ({ image, header, subtitle }) => (
  <TouchableOpacity style={styles.container}>
    <Image
      source={image && image.toString().startsWith('http') ? {uri: image} : image}
      style={styles.image}
    />
    <Text style={styles.title} ellipsizeMode={'tail'} numberOfLines={1}>{header}</Text>
    <Text style={styles.subtitle} ellipsizeMode={'tail'} numberOfLines={1}>{subtitle}</Text>
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 10
  },
  image: {
    width: 128,
    height: 128
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    marginTop: 10
  },
  subtitle: {
    fontSize: 18
  }
})

Slide.propTypes = {
  image: PropTypes.any.isRequired,
  header: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}

export default Slide
