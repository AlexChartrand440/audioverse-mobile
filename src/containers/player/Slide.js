import React from 'react'
import PropTypes from 'prop-types'
import { View, TouchableOpacity, Image, Text, StyleSheet } from 'react-native'

const Slide = ({ image, header, subtitle, onPress }) => (
  <View style={styles.container}>
    <Image
      source={image && image.toString().startsWith('http') ? {uri: image} : image}
      style={styles.image}
    />
    <TouchableOpacity style={styles.center} onPress={onPress}>
      <Text style={styles.title} ellipsizeMode={'tail'} numberOfLines={1}>{header}</Text>
      <Text style={styles.subtitle} ellipsizeMode={'tail'} numberOfLines={1}>{subtitle}</Text>
    </TouchableOpacity>
  </View>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 10
  },
  center: {
    alignItems: 'center'
  },
  image: {
    width: 128,
    height: 128,
    borderRadius: 64
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    marginTop: 10
  },
  subtitle: {
    fontSize: 18
  }
})

Slide.propTypes = {
  image: PropTypes.any.isRequired,
  header: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  onPress: PropTypes.func
}

export default Slide
