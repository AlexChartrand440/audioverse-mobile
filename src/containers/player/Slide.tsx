import React from 'react'
import {
  Image,
  StyleSheet,
  TouchableOpacity,
  View,
} from 'react-native'

// eslint-disable-next-line @typescript-eslint/no-var-requires
const MarqueeText: any = require('react-native-marquee').default

interface Props {
  image: any
  header: string
  subtitle: string
  onPress: () => void
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 10,
  },
  center: {
    alignItems: 'center',
  },
  image: {
    width: 128,
    height: 128,
    borderRadius: 64,
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    marginTop: 10,
  },
  subtitle: {
    fontSize: 18,
  },
})

const Slide: React.FC<Props> = ({ image, header, subtitle, onPress }) => {
  return (
    <View style={styles.container}>
      <Image
        source={image && image.toString().startsWith('http') ? {uri: image} : image}
        style={styles.image}
      />
      <TouchableOpacity style={styles.center} onPress={onPress}>
        <MarqueeText marqueeOnStart duration={3500} loop style={styles.title}>{header}</MarqueeText>
        <MarqueeText marqueeOnStart duration={3500} loop style={styles.subtitle}>{subtitle}</MarqueeText>
      </TouchableOpacity>
    </View>
  )
}

export default Slide
