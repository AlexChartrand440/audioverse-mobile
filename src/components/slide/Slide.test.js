import React from 'react'
import Slide from './Slide'
import defaultImage from '../../../assets/av-logo.png'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<Slide image={defaultImage} header="" />).toJSON()
  expect(tree).toMatchSnapshot()
})
