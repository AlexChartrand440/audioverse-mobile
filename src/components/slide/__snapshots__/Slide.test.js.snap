// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 128,
        "width": 128,
      }
    }
  />
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "fontSize": 26,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
</View>
`;
